### PR TITLE
fix(stft_loss): mask frequency axis (not frames) for high-band loss

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
## Summary

With `band="high"` (used by `configs/DNS-large-high.json`), `STFTLoss.forward()` is meant to compute the spectral-convergence and log-magnitude losses on the upper half of the frequency bins only ("only select high frequency bands"). Instead it drops the second half of the time frames and keeps all frequencies, so the high-band loss is computed over low frequencies and half the utterance.

## Root cause

`stft()` returns the magnitude spectrogram as `(B, #frames, fft_size // 2 + 1)` (per its docstring: real/imag `(B, freq, frames)` are transposed with `.transpose(2, 1)`). The high-band code used axis 1 (frames) for the frequency mask:

```python
freq_mask_ind = x_mag.shape[1] // 2  # only select high frequency bands
sc_loss  = self.spectral_convergence_loss(x_mag[:,freq_mask_ind:,:], y_mag[:,freq_mask_ind:,:])
mag_loss = self.log_stft_magnitude_loss(x_mag[:,freq_mask_ind:,:], y_mag[:,freq_mask_ind:,:])
```

## Fix

Mask the last axis (frequency bins):

```python
freq_mask_ind = x_mag.shape[-1] // 2
... x_mag[...,freq_mask_ind:] ...
```

## Testing

No test suite exists in the repo. Verified with a standalone script that records the tensor shapes seen by both sub-losses for a `band="high"` `STFTLoss(fft_size=64)` on random signals (full magnitude shape `(2, 63 frames, 33 bins)`):

- With the fix: losses receive `(2, 63, 17)` — frames intact, upper half of the 33 frequency bins selected (PASS).
- With the fix stashed (unmodified code): losses receive `(2, 32, 33)` — frames wrongly halved, all frequencies kept (AssertionError).

Command: `uv run --python 3.11 --with torch --index-url https://download.pytorch.org/whl/cpu python test_high_band.py` (CPU torch, no GPU needed).

## Why existing tests missed it

The repo has no tests, and the bug is silent: the loss still computes a finite scalar, just over the wrong axes.

### CODE REVIEW

**Comment by andrewwhitecdw** [comment]
Closing this sweep-generated PR: PR has 2 commits; sweep requires exactly one commit per PR. It does not meet the sweep requirements (single signed-off commit).
